File: src/main/kotlin/mx/com/team9/domain/Cuenta.java
package mx.com.team9.domain;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Cuenta {
    private String idCuenta;
    private double saldo;

    public List<Movimiento> getListaMovimientos() {
        return listaMovimientos;
    }

    public void setListaMovimientos(List<Movimiento> listaMovimientos) {
        this.listaMovimientos = listaMovimientos;
    }

    private String nombre;
    private List<Movimiento> listaMovimientos;
    private Usuario usuario;
    private LocalDateTime fechaCreacion;

    public Cuenta(String idCuenta, double saldo, String nombre) {
        this.idCuenta = idCuenta;
        this.saldo = saldo;
        this.nombre = nombre;
        this.listaMovimientos = new ArrayList<Movimiento>();
        this.fechaCreacion = LocalDateTime.now();
    }

    public void consultaSaldo() {
        System.out.println("EL SALDO DE LA CUENTA ES: " + saldo);
    }

    public List<Movimiento> obtenerMovimientos() {
        return listaMovimientos;
    }

    // getters y setters
    public String getIdCuenta() {
        return idCuenta;
    }

    public void setIdCuenta(String idCuenta) {
        this.idCuenta = idCuenta;
    }

    public double getSaldo() {
        return saldo;
    }

    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }

    public LocalDateTime getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(LocalDateTime fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }
}

File: src/main/kotlin/mx/com/team9/domain/Cuenta.kt
package mx.com.team9.domain

import java.time.LocalDateTime

/*  mx.com.team9.models.Cuenta.kt tendra los atributos para poder manejar las cuentas de los usuarios y las
    operaciones que se pueden realizar en ellas.

    Posible atributos:
    - Numero de cuenta
    - Saldo
    - Tipo de cuenta
    - Fecha de creacion
    - Estado de la cuenta
    - Usuario
    - Movimientos

    */


class Cuenta(
    val idCuenta: String,
    //TODO: AGREGAR NOMBRE DE CUENTA
    var saldo: Double,
    var nombre: String,
) {
    val listaMovimientos =  mutableListOf<Movimiento>()

    lateinit var usuario: Usuario

    val fechaCreacion: LocalDateTime = LocalDateTime.now()

    fun consultaSaldo() {
        println("El saldo de la cuenta es: $saldo")
    }

    // consultar los movimientos de la cuenta
    fun obtenerMovimientos(): List<Movimiento> {
        return listaMovimientos
    }
}
